tamr_client/dataset/dataframe.py

```diff
@@ -8,7 +8,7 @@
 import pandas as pd
 
 from tamr_client.types import JsonDict
-from tamr_client.dataset import Dataset
+from tamr_client.dataset.dataset import Dataset
 from tamr_client.session import Session
 from tamr_client.dataset import dataset
```

Comment:
This is importing the `dataset.py` module from the `tamr_client.dataset` package via import hoisting in `tamr_client/dataset/__init__.py`.

To import the `dataset.py` module without using import hoisting:

```suggestion
import tamr_client.dataset.dataset as dataset
```

Same goes for the `record` import below this one and other imports in other files in this PR.